### PR TITLE
Split "God"-class Helper

### PR DESCRIPTION
**Pull request by pk-work**

### CODE REVIEW

File: src/test/java/io/neonbee/test/base/ODataRequest.java [hunk 1]
```diff
@@ -101,20 +101,22 @@ public ODataRequest setBody(Buffer body) {
...
+
+            WebClientOptions clientOpts = new WebClientOptions().setDefaultHost("localhost").setDefaultPort(port);
+            HttpRequest<Buffer> httpRequest = WebClient.create(vertx, clientOpts).request(method, basePath + getUri());
+
+            httpRequest.queryParams().addAll(query);
+            httpRequest.putHeaders(headers);
+            Optional.ofNullable(interceptor).ifPresent(i -> i.accept(httpRequest));
+            httpRequest.queryParams(); // ensures that query params are encoded
+
+            return Optional.ofNullable(body).map(b -> httpRequest.sendBuffer(b)).orElse(httpRequest.send());
```
Comment (SAPDaniloWork) [inline src/test/java/io/neonbee/test/base/ODataRequest.java:118]: why not with method reference
`return Optional.ofNullable(body).map(httpRequest::sendBuffer).orElse(httpRequest.send());`

File: src/test/java/io/neonbee/test/helper/DeploymentHelper.java [hunk 2]
```diff
@@ -72,7 +73,7 @@
             }
         }
 
-        return CompositeFuture.all(Helper.uncheckedMapper(undeployFutures)).mapEmpty();
+        return CompositeFuture.all(uncheckedMapper(undeployFutures)).mapEmpty();
     }
 
     public static boolean isVerticleDeployed(Vertx vertx, Class<? extends Verticle> verticleToCheck) {
```
Comment (SAPDaniloWork) [inline src/test/java/io/neonbee/test/helper/DeploymentHelper.java:79]: unrelated: the last filter statement can be simplified:
```
return vertx.deploymentIDs().stream()
                .flatMap(deploymentId -> ((VertxInternal) vertx).getDeployment(deploymentId).getVerticles().stream())
                .anyMatch(verticleToCheck::isInstance);
```

File: src/main/java/io/neonbee/internal/helper/FileSystemHelper.java [hunk 3]
```diff
@@ -77,6 +89,40 @@
...
+     * @param vertx The related Vert.x instance
+     * @param path  The path of the file
+     * @return a future to the converted {@link JsonObject}
+     */
+    public static Future<JsonObject> readJSON(Vertx vertx, Path path) {
+        return readFile(vertx, path).map(Buffer::toJsonObject);
+    }
+
+    /**
+     * This method reads the entire file in YAML format asynchronously and converts it to a {@link JsonObject}. * @param
```
Comment (SAPDaniloWork) [inline src/main/java/io/neonbee/internal/helper/FileSystemHelper.java:104]: java doc c&p mistake:
```
* @param
     * vertx The related Vert.x instance
```

File: src/main/java/io/neonbee/internal/helper/BufferHelper.java [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+        }
+
+        /**
+         * Get the current position of the buffer.
+         *
+         * @return the current position of the buffer
+         */
+        public int getPosition() {
+            return position;
+        }
```
Comment (SAPDaniloWork) [inline src/main/java/io/neonbee/internal/helper/BufferHelper.java:91]: code is never used

**Review by SAPDaniloWork — COMMENTED** [review]
a rough review as discussed

File: src/main/java/io/neonbee/internal/helper/CollectionsHelper.java [hunk 5]
```diff
@@ -0,0 +1,174 @@
...
+     *
+     * @param <T>               the type of the collection
+     * @param <C>               the collection to return
+     * @param collection        the collection to copy
+     * @param collectionFactory the factory to create the collection to add the copied items into
+     * @return a mutable deep copy of the given collection
+     */
+    public static <T, C extends Collection<T>> C mutableCopyOf(C collection, Supplier<C> collectionFactory) {
+        return Optional.ofNullable(collection).map(Collection::stream).orElseGet(Stream::empty)
+                .map(CollectionsHelper::copyOf).collect(Collectors.toCollection(collectionFactory));
```
[the remark this answers is not in the document]
Comment (pk-work) [reply src/main/java/io/neonbee/internal/helper/CollectionsHelper.java:66]: We save some characters here, but the logic is less intuitive at least for me. So I will not change this.

File: src/main/java/io/neonbee/internal/helper/CollectionsHelper.java [hunk 6]
```diff
@@ -0,0 +1,174 @@
...
+     * Create mutable deep copy of a set by creating a copy of all mutable items and adding them to a new set.
+     *
+     * @param <T> the type of the set
+     * @param set the set to copy
+     * @return a mutable deep copy of the given set
+     */
+    public static <T> Set<T> mutableCopyOf(Set<T> set) {
+        return Optional.ofNullable(set).map(Set::stream).orElseGet(Stream::empty).map(CollectionsHelper::copyOf)
+                .collect(Collectors.toCollection(HashSet::new));
+    }
```
[the remark this answers is not in the document]
Comment (pk-work) [reply src/main/java/io/neonbee/internal/helper/CollectionsHelper.java:52]: We save some characters here, but the logic is less intuitive at least for me. So I will not change this.

File: src/main/java/io/neonbee/internal/helper/CollectionsHelper.java [hunk 7]
```diff
@@ -0,0 +1,174 @@
...
+     *
+     * @param <K> the key type
+     * @param <V> the value type
+     * @param map the map to copy
+     * @return a mutable deep copy of the given map
+     */
+    public static <K, V> Map<K, V> mutableCopyOf(Map<K, V> map) {
+        return Optional.ofNullable(map).map(Map::entrySet).map(Set::stream).orElseGet(Stream::empty)
+                .collect(Collectors.toMap(entry -> copyOf(entry.getKey()), entry -> copyOf(entry.getValue()),
+                        (valueA, valueB) -> valueB, NullLiberalMergingHashMap::new));
```
[the remark this answers is not in the document]
Comment (pk-work) [reply src/main/java/io/neonbee/internal/helper/CollectionsHelper.java:80]: We save some characters here, but the logic is less intuitive at least for me. So I will not change this.

File: src/test/java/io/neonbee/test/helper/DeploymentHelper.java (hunk 2 above)
Comment (pk-work) [reply src/test/java/io/neonbee/test/helper/DeploymentHelper.java:79]: Done, good catch

**Pull request merged**